instructions/steps.md

```diff
@@ -1023,488 +1025,62 @@ This is a slice notation.
-    //# identifying an an AWS KMS multi-Region ARN (Section 2.8) called with this
-    //# input.
-    ensures "arn:" <= identifier
-      ==>
-        var arnIdentifier := ParseAwsKmsIdentifier(identifier).value;
-        MultiRegionAwsKmsIdentifier?(arnIdentifier) == MultiRegionAwsKmsArn?(arnIdentifier.a)
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If the input starts with "alias/", this an AWS KMS alias and
-    //# not a multi-Region key id and MUST return false.
-    ensures "alias/" <= identifier
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier).value;
-        !MultiRegionAwsKmsIdentifier?(ParseAwsKmsIdentifier(identifier).value)
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If the input starts
-    //# with "mrk-", this is a multi-Region key id and MUST return true.
-    ensures "mrk-" <= identifier
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier).value;
-        MultiRegionAwsKmsIdentifier?(resource)
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If
-    //# the input does not start with any of the above, this is not a multi-
-    //# Region key id and MUST return false.
-    ensures
-        && !("arn:" <= identifier )
-        && !("alias/" <= identifier )
-        && !("mrk-" <= identifier )
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier);
-        !MultiRegionAwsKmsIdentifier?(resource.value)
-  {}
+In our makefile there are targets for each supported runtime.
+Feel free to play around with other values!
 
+```bash
+make execute
 ```
 
-## Step 21
-
-Let's run Duvet one last time:
+If you would like to investigate the source that Dafny produces
+this will output the source to a `runtimes` directory.
 
 ```bash
-make duvet_report
+make compile
 ```
 
-Success! The command no longer fails
-(if you don't believe the lack of error messages, you can check with `echo $?`)
-and the report is a happy sea of green.
-Congratulations on your formally-verified implementation!
-
-## Extra Credit
+## More Extra Credit
 
 Here are a couple of bonus questions if you are hungry for more:
 
+1. Head over [here](next-steps.md) if you want to continue
+  and complete the rest of the specification.
+  This is a guided tour and has a few more Dafny goodies.
+
 1. If you have strong instincts to keep your code DRY,
    you might feel uncomfortable with all of the repeated `0 < | ... |`
    expressions everywhere we require a non-empty string.
    Try defining a `NonEmptyString` subset type and use it
    in place of `string` wherever it makes sense to tighten things up.
    Where should the relevant Duvet citations move to?
-   
-2. Now that we have implemented parsing, 
+
+1. Now that we have implemented parsing, 
    try implementing the other direction in a `AwsKmsIdentifierToString`
    function and write a lemma to prove that it is always
    the inverse of `ParseAwsKmsIdentifier`.
```

Comment:
Needs to be `AwsKmsArnToString` instead since we only got that far (and I think we should specify that the bonus questions don't have to be done in order).

Reply:
needed to fix other things as well. (index.dfy)